src/elements/name_section.rs

```diff
@@ -10,92 +10,156 @@ const NAME_TYPE_LOCAL: u8 = 2;
 
 /// Debug name information.
 #[derive(Clone, Debug, PartialEq)]
-pub enum NameSection {
-	/// Module name section.
-	Module(ModuleNameSection),
-
-	/// Function name section.
-	Function(FunctionNameSection),
-
-	/// Local name section.
-	Local(LocalNameSection),
-
-	/// Name section is unparsed.
-	Unparsed {
-		/// The numeric identifier for this name section type.
-		name_type: u8,
-		/// The contents of this name section, unparsed.
-		name_payload: Vec<u8>,
-	},
+pub struct NameSection {
+	/// Module name subsection.
+	module_name_subsection: Option<ModuleNameSubsection>,
+
+	/// Function name subsection.
+	function_name_subsection: Option<FunctionNameSubsection>,
+
+	/// Local name subsection.
+	local_name_subsection: Option<LocalNameSubsection>,
+}
+
+impl NameSection {
+	/// Creates a new name section.
+	pub fn new(module_name_subsection: Option<ModuleNameSubsection>,
+			   function_name_subsection: Option<FunctionNameSubsection>,
+			   local_name_subsection: Option<LocalNameSubsection>) -> Self {
+		Self {
+			module_name_subsection,
+			function_name_subsection,
+			local_name_subsection,
+		}
+	}
+
+	/// Module name subsection of this section.
+	pub fn module_name_subsection(&self) -> Option<&ModuleNameSubsection> {
```

Comment:
name name could be just "module", there other is clear from signature, how do you think?

so 
`pub fn module(&self) -> Option<&ModuleNameeSubsection>`
?

Same with others